### ISSUE TITLE
`onChange` event on radio

### ISSUE THREAD

**Issue by yankouskia** [facebook/react#8727]
Could you please, explain this behaviour?

Here: https://jsfiddle.net/69z2wepo/67116/ - clicking on radio `always` emits `onClick` and `onChange` events

Here is example without `React`: https://jsfiddle.net/8mnjtt8w/4/ - clicking on radio emits only `onClick` event.

Thank you!

**Comment by aweary** [comment]
@yankouskia React special cases `onChange`, meaning it doesn't necessarily correspond to the DOM `change` event, in order to make it easier to handle inputs. For example, `change` events usually occur for text inputs on `blur`, but most people expect them to occur when the input's value is actually changed, so we normalize that. You can see this behavior [documented here](https://facebook.github.io/react/docs/dom-elements.html#onchange)

Now, that said, this is still unexpected behavior since the input value isn't changed. This was already reported in https://github.com/facebook/react/issues/1471 and resolved in https://github.com/facebook/react/pull/5746. But https://github.com/facebook/react/pull/5746 has not been released yet. We're hoping to get it released soon thanks to the backport PR opened at https://github.com/facebook/react/pull/8575

**tl;dr**: we normalize input events in general, but this radio input behavior is wrong and already fixed. We're working on getting it released 👍

**Comment by yankouskia** [comment]
@aweary , thank you! 👍

**Comment by wangcongyi** [comment]
@aweary  the lastest version 15.4.2 does not seem to fix the bug ，how to solve the old version of it?
